File: leetcode/lc-0014/solution.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if (strs.empty()) return "";
        int n = strs.size();
        if (n == 1) return strs[0];
        string prefix = strs[0];
        for (int i = 0; i < prefix.size(); i++) {
            for (int j = 1; j < n; j++) {
                if (strs[j].size() <= i) {
                    return string(prefix.begin(), prefix.begin() + i);
                }
                if (strs[j][i] != prefix[i]) {
                    return string(prefix.begin(), prefix.begin() + i);
                }
            }
        }
        return prefix;
    }
};

File: leetcode/lc-0014/solution.rs
impl Solution {
    pub fn longest_common_prefix(strs: Vec<String>) -> String {
        if strs.is_empty() {
            return "".to_string();
        }

        let n = strs.len();
        if n == 1 {
            return strs[0].clone();
        }

        let prefix = strs[0].clone();
        for i in 0..prefix.len() {
            for j in 1..n {
                if strs[j].len() <= i {
                    return prefix.chars().take(i).collect();
                }
                if strs[j].as_bytes()[i] != prefix.as_bytes()[i] {
                    return prefix.chars().take(i).collect();
                }
            }
        }
        return prefix;
    }
}
